### ISSUE TITLE
No possibility to change retry policy

### ISSUE THREAD

**Issue by MartinSeeler** [thinkaurelius/titan#524]
It should be possible to change the `FixedRetryBackoffStrategy` to something other. I know it's in your todo, since it`s mentioned [in this line](https://github.com/thinkaurelius/titan/blob/master/titan-cassandra/src/main/java/com/thinkaurelius/titan/diskstorage/cassandra/astyanax/AstyanaxStoreManager.java#L354), but it is annoying to get an error each 1000ms without the possibility to try and catch this connection problem or set an end after 5 retries, particularly if the connection to the backend is lost, e.g. through a Cassandra shutdown.

**Comment by dalaro** [comment]
There's confusing terminology here.  Astyanax distinguishes between RetryPolicy and RetryBackoffStrategy.  They sound and look similar, but they are separate interfaces without any inheritance relationship and they have different functions.

RetryPolicy is used for retrying operations (e.g. get_slice).  This is already configurable in the Titan Astyanax manager.

The source line you linked instantiates a RetryBackoffStrategy.  This controls how frequently Astyanax tries to talk to an uncommunicative host.  Well, it theoretically does.  But I vaguely recall some surprising behavior in astyanax-core last time I looked into the mechanics.  For instance, I still find no references to RetryBackoffStrategy.Instance.suspend() in quick search of Astyanax's master branch.

Anyway, I made this configurable: 5005da5a3b85280f95f175aa04f7015c94decaf1.  Now it's a possible to use either of Astyanax's default RetryBackoffStrategy implementations or a custom implementation, provided that the custom implementation has a no-arg or int-arg-only constructor.
